File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/status/get_error_message.py
import humanoid_sdk_py
from humanoid_sdk_py import h10w


h10w_status = h10w.H10wStatus()
print("=== 获取机器人当前运动状态信息 ===")

ret, error_msg = h10w_status.getErrorMessage()
if ret != 0 or error_msg is None:
    print("获取错误信息失败，任务终止")
    exit(-1)

print("错误代码：", error_msg.error_code)
print("错误ID：", error_msg.id)
print("错误等级：", error_msg.level)
print("模块ID：", error_msg.module)
print("错误信息：", error_msg.msg)

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/status/get_error_message.cpp
// get_error_message.cpp

#include <iostream>
#include <iomanip>
#include "h10-w/h10w_status.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 获取机器人当前错误信息示例 ===" << std::endl;

  H10wStatus status;
  ErrorMessageParams errorMessage;

  int result = status.getErrorMessage(errorMessage);
  if (result != 0) {
    std::cerr << "获取机器人当前错误信息失败: " << std::endl;
    return -1;
  }
  std::cout << "错误码ID: " << errorMessage.id << std::endl;
  std::cout << "错误等级: " << static_cast<int>(errorMessage.level)
            << std::endl;
  std::cout << "错误模块: " << static_cast<int>(errorMessage.module)
            << std::endl;
  std::cout << "错误代码: " << errorMessage.error_code << std::endl;
  std::cout << "错误信息: " << errorMessage.msg << std::endl;
  return 0;
}